src/Components/Admin/ObjectEditor/object.js

```diff
@@ -0,0 +1,318 @@
+import React, { useEffect, useState } from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import { useErrorHandler } from "react-error-boundary";
+import Select from "react-select";
+import AddIcon from "@material-ui/icons/Add";
+import { Button, IconButton } from "@material-ui/core";
+import {
+    DeleteForever,
+    KeyboardArrowDown,
+    KeyboardArrowUp,
+} from "@material-ui/icons";
+
+import { getPuzzle, editPuzzle } from "../../../lib/puzzleEndpoints";
+
+const useStyles = makeStyles((theme) => ({
+    page: {
+        marginTop: theme.spacing(8),
+    },
+    sceneAndTransitionContainer: {
+        minWidth: 600,
+    },
+    introContainer: {
+        minWidth: 850,
+        display: "flex",
+    },
+    container: {
+        paddingTop: theme.spacing(12),
+    },
+    dragAndDropContainer: {
+        display: "flex",
+        alignItems: "center",
+        marginLeft: "16px",
+    },
+    emptyButtonsContainer: {
+        display: "flex",
+        width: "100%",
+        flexDirection: "column",
+        flexGrow: 1,
+        justifyContent: "center",
+        alignItems: "space-around",
+    },
+    buttonContainer: {
+        display: "flex",
+        width: "100%",
+        justifyContent: "center",
+        marginTop: theme.spacing(12),
+        marginBottom: theme.spacing(4),
+    },
+    button: {
+        display: "flex",
+        justifyContent: "center",
+        padding: "16px 12px",
+        backgroundColor: "#E2E5ED",
+        borderRadius: "12px",
+    },
+}));
+
+class TextPaneView extends React.Component {
```

Comment:
Why are we using class components? We should be using functional components to align with the rest of the code base

Reply:
Could we also move this component into another file? Would be bit easier to maintain

Reply:
TextPaneView needs to call ObjectEditor's saveTexts function, is it possible to do this if it were a functional component, if so, how would I do it?

Reply:
I suggest making a wrapper class that wraps ObjectEditor and TextPaneView and create functions in the wrapper class that can be passed to both components.

Reply:
Let's move to separate file. I'm okay with this being a class component - they work together so not a big deal.

In the future, let's do functional components in the cdc-frontend repo. I know aframe-inspector is outdated and still uses class components, so you can keep using class components there.